src/LuaTransformer.ts

```diff
@@ -593,6 +592,7 @@ export class LuaTransformer {
                 statement,
                 className,
                 localClassName,
+                classNameText,
```

Comment:
Yes, they are actually 2 different things. `localClassName` is the local variable used when setting up an exported class, and it will have a 'safe' name. `classNameText` is the display-name of the class and won't be manipulated at all.